variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/filter/FilterBuilder.java

```diff
@@ -96,11 +109,72 @@ public FilterBuilder withVariantTypes(List<VariantType> types) {
         return this;
     }
 
+    public FilterBuilder withVariantTypes(VariantType variantType){
+        if(variantType!=null){
+            filters.add(new VariantRepositoryTypeFilter(new ArrayList<>(Arrays.asList(variantType))));
+        }
+        return this;
+    }
     public FilterBuilder withAlternates(List<String> alternates) {
         if (alternates != null && !alternates.isEmpty()) {
             filters.add(new VariantRepositoryAlternateFilter(alternates));
         }
         return this;
     }
 
+    public FilterBuilder withAlternates(String alternate){
+        if(alternate!=null){
+            filters.add(new VariantRepositoryAlternateFilter(new ArrayList<>(Arrays.asList(alternate))));
+        }
+        return this;
+    }
+
+    public FilterBuilder withReferenceBases(String referenceBases){
+        if(referenceBases!=null){
+            filters.add(new VariantRepositoryReferenceBasesFilter(new ArrayList<>(Arrays.asList(referenceBases))));
+        }
+        return this;
+    }
+
+    public FilterBuilder withStart(Long start){
+        if(start!=null){
+            filters.add(new VariantRepositoryStartFilter(start,RelationalOperator.EQ));
+        }
+        return this;
+    }
+
+    public FilterBuilder withStartMin(Long startMin){
```

Comment:
Can you please check if there is any way you can use the existing `Region` class to deal with these positional ranges?

Reply:
I guess!
I could create 2 Region objects like **regionMin**(with start=startMin and end=endMin) and **regionMax** (start=startMax and end=endMax) and can use this (But the chromosome data member would actually become redundant here) 

Or I could just add 4 more datamembers(startMax, endMax, startMin and endMin) to the **Region** class  and can use this for filtering.